### ISSUE TITLE
KR-29 starting GEOS from fresh extract

### ISSUE THREAD

**Issue by lighth7015** [bluewaysw/pcgeos#79]
When starting any of the executables (`go.bat`, `loader.com`) on FreeDOS, I receive the following message-
```
No valid font files found in USERDATA\FONT directory. (KR-29)
```

How do I resolve this issue?

**Comment by hanslse** [comment]
Under [system] section look for:

fs = ms4 geo

Change to:

fs = os2.geo
primaryFSD = os2.geo

Note: there are spaces before and after the equal to sign

Hans

**Comment by hanslse** [comment]
Sorry, I forgot to say that you should edit geos.ini in the ensemble folder. Check for the system section and edit like I propose

**Comment by lighth7015** [comment]
It's still giving me the `KR-29` error code.

**Comment by lighth7015** [comment]
I should note that I'm using linux and `dosemu` (KVM) as my VM.

**Comment by jirkakunze** [comment]
Which version of PC/Geos do you use? If you've created a version based on the GitHub sources, the font files are still missing.

**Comment by lighth7015** [comment]
@jirkakunze I downloaded the ZIP from the website and extracted it.

**Comment by jirkakunze** [comment]
Ok, then you have downloaded BreadBox Ensemble Verions 4.13 from bluewaysw.de. Please check the content of the directory ensemble/userdata/font. There must be 15 *.fnt files there. The error message indicates that these files are missing.

**Comment by lighth7015** [comment]
I count 15 files.

**Comment by jirkakunze** [comment]
I played a little with dosemu. Here's what works for me:

Edited config.sys
files=120
buffers=40

Edit geos.ini:
fs = msnet.geo

**Comment by lighth7015** [comment]
woot! that fixed it Now to install the other software! :D

**Comment by lighth7015** [comment]
Does the SDK let me compile from within Ensemble?

**Comment by bolle732** [comment]
Hello Robert

It depends also if you're using short file names (SFN) or long file names (LFN).

- With SFN under DOSEMU2
Disable LFN in DOSEMU with the setting "$_lfn_support = (off)".

Then edit your "geos.ini" as follow:
[system]
fs = msnet.geo

- With LFN under DOSEMU2
Activate LFN in DOSEMU with the setting "$_lfn_support = (on)".

Then edit your "geos.ini" as follow:
[system]
fs = msnet.geo
primaryFSD = mslf.geo

Hope this helps.
Andreas

On Saturday, March 21, 2020 20:13 CET, Robert Butler <notifications@github.com> wrote:
 
 
@jirkakunze I downloaded the ZIP from the website and extracted it.

**Comment by andrewbird** [comment]
@bolle732 I'm interested in the reason why if not using LFN in GEOS you need to disable LFN in Dosemu2? I'd have thought SFN functions would be unchanged even with `$_lfn_support = (on)`. What effect do you see?

**Comment by jirkakunze** [comment]
@lighth7015  No, with the SDK loaded from bluewaysw.de you can develop applications, libraries and drivers for PC/Geos under Windows. For this you need Borland C++.  With the SDK builded from GitHub sources, you can run Windows and Linux on your developer machine. This SDK uses Watcom C++.

**Comment by bolle732** [comment]
Hello Andrew

Maybe the step to disable LNF in DOSEMU2 is not needed. I only have it recorded in my snippets to be sure that I get a reproducible setup if testing with SNF and LFN. Over the last 2 or 3 years, there have been so many changes regarding the configuration in DOSEMU2, that it is difficult to compare different tests and installations...

Andreas

On Monday, March 23, 2020 10:11 CET, Andrew Bird <notifications@github.com> wrote:
 
 
@bolle732 I'm interested in the reason why if not using LFN in GEOS you need to disable LFN in Dosemu2? I'd have thought SFN functions would be unchanged even with $_lfn_support = (on). What effect do you see?

**Comment by andrewbird** [comment]
Hi Andreas,
   Thanks for letting me know.

**Comment by lighth7015** [comment]
Ah, linux. I will do that then.

**Comment by lighth7015** [comment]
So how do I compile natively under linux? The build system only seems to want to compile for win32.

**Comment by jirkakunze** [comment]
Checkout the sources form our GitHub repository, install Watcom c++ and follow the instructions on https://github.com/bluewaysw/pcgeos.

**Comment by lighth7015** [comment]
Yes, but per the instructions, Watcom generates win32 binaries. I'm not on Windows. I mean if that's the way it has to be then I guess I'll just run WINE; but I would prefer native ELF instead of PE build tools is all I'm saying. :D

**Comment by frehwagen** [comment]
WATCOM should generate binaries for both in the same run, following the instructions in the README. PATH setup needs to be translated for linux of course.

You can follow a full build on a linux machine here:
https://travis-ci.com/github/bluewaysw/pcgeos

Build #74 for instance
line 9119ff
--- Tools/glue:linux.md/glue. ---

**Comment by lighth7015** [comment]
Yeah, I saw that! So do the C examples build at all?

**Comment by frehwagen** [comment]
Proved for the ones that are part of the overall ensemble build (pcgeos/Installed/Makefile), there are a lot of C modules in there.  Proved to build, working currently on proving that they all are executable. There are also many C project that has been migrated that are not part of the ensemble Makefile.

Do you have examples you would like to try?

**Comment by krasnit** [comment]
Also having the same problem K-29 of no fonts found trying to run Ensemble 4.1.3 in DOSBOX on Windows 10.  The problem with DOSBOX is that there is no config.sys file, so one assumes that right number of Buffers and Files have been set up by default.  Any help would be appreciated.

DOSBOX can be run full screen with a few tweaks, so that is why am trying Ensemble with it.  I had no luck whatsoever trying to run Ensemble with FreeDoos running as a VM in Oracle's VirtualBox.

**Comment by krasnit** [comment]
Looks like I solved the problem.  fs = os2.geo  in GEOS.INI works.  As for the USB mouse connected to this computer, selecting the PS/2 mouse in Ensemble works for it.

Although full scrreen, I still cannot get the 16:9 or 16:10 screen format.  Only 4:3.  

Now to get networking or Internet on this .......
